packit_service/worker/handlers/forges.py

```diff
@@ -64,11 +64,22 @@ def run(self) -> TaskResults:
         match.
         :return: TaskResults
         """
-        GithubInstallationModel.create(event=self.installation_event)
+        # if there is already a record in the DB, the app was reinstalled
+        is_reinstalled = (
+            GithubInstallationModel.get_by_account_login(
+                self.installation_event.account_login
+            )
+            is not None
+        )
+        GithubInstallationModel.create_or_update(event=self.installation_event)
         # try to add user to allowlist
         allowlist = Allowlist(self.service_config)
-        if not allowlist.add_namespace(
-            f"github.com/{self.account_login}", self.sender_login
+        namespace = f"github.com/{self.account_login}"
+        # if the namespace was not on our allowlist (in any state) or
+        # the app was reinstalled and the namespace was not approved,
+        # create an issue in notifications repo
+        if not allowlist.add_namespace(namespace, self.sender_login) or (
+            is_reinstalled and not allowlist.is_approved(namespace)
         ):
```

Comment:
I'm confused because I understand the `if` statement as "if the namespace is reinstalled and not approved, create a new issue", which makes perfect sense to me (alternatively we could recycle the old issue but could become messy)